testdata/fields_with_default_types.txtar

```diff
@@ -0,0 +1,44 @@
+-- cue --
+
+#Type: "a" | "b" | "c" @cuetsy(kind="type")
+
+#Base: {
+    valueType:   #Type
+    defaultType: #Type
+} @cuetsy(kind="interface")
+
+#StructWithDefaults: {
+    #Base
+    baseType:              #Type
+    valueType:             #Type
+    defaultType:           #Type & (*"a" | _)
+    noOverrideDefaultType: #Type & (*"b" | _)
+    noOverrideType:        #Type
+    defaultString:         #Type | *"invalid"
+    defaultNumber:         #Type | "something" | *34
+} @cuetsy(kind="interface")
+
+-- ts --
+
+export type Type = ('a' | 'b' | 'c');
+
+export interface Base {
+  defaultType: Type;
+  valueType: Type;
+}
+
+export interface StructWithDefaults extends Base {
+  baseType: Type;
+  defaultNumber: Type;
+  defaultString: Type;
```

Comment:
same deal as in the other PR - as CUE sees it, the inclusion of the concrete default value increases the space of valid values by exactly that value. We have to reflect that on the TS side.

```suggestion
  defaultNumber: (Type | 34);
  defaultString: (Type | 'invalid');
```